src/camvidlog2/vid.py

```
@@ -410,3 +425,76 @@ def slice_frame(
+    if slice_overlap < 0.0:
+        raise ValueError("slice overlap must not be negative")
+    if slice_overlap >= 1.0:
+        raise ValueError("slice overlap must not be greater or equal to 1.0")
+
+    # get the dimensions of the input image
+    frame_height, frame_width = frame.shape[:2]
+
+    if slice_width > frame_width:
+        raise ValueError("slice width must be no wider than frame")
+    if slice_height > frame_height:
+        raise ValueError("slice width must be no higher than frame")
```

Ruff (0.11.9)

Avoid specifying long messages outside the exception class

(TRY003)